### ISSUE TITLE
Add support for Fine-tuning API

### ISSUE THREAD

**Issue by KacperKluka** [nomtek/mistralai_client_dart#57]
**Is your feature request related to a problem? Please describe.**
Mistral AI introduced the possibility to fine-tune their LLMs to even further increase their accuracy and performance. It was added to the JS client in [version 0.5.0](https://github.com/mistralai/client-js/releases/tag/0.5.0) and we should also support it to offer the full functionality of the Mistral AI API.

**Describe the solution you'd like**
Implement the "files" and "jobs" features based on the official JS client.

**Describe alternatives you've considered**
N/A

**Additional context**
Mistral docs - https://docs.mistral.ai/capabilities/finetuning/
Mistral JS client implementation - https://github.com/mistralai/client-js/commit/0d1cc9c9cf4b56feac057aecc0f9c1ec60d631dd
